### PR TITLE
[11.x] Allow Carbon 2 and 3

### PR DESCRIPTION
**Pull request by kylekatarnls**
Upgrade Carbon to v3 for projects not having dependencies requiring v2 with still allowing to install v2.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -36,7 +36,7 @@
         "league/commonmark": "^2.2.1",
         "league/flysystem": "^3.8.0",
         "monolog/monolog": "^3.0",
-        "nesbot/carbon": "^2.67",
+        "nesbot/carbon": "^2.72.2 || ^3.0.0-beta.3@beta",
```
Comment (driesvints) [inline composer.json:39]:
```suggestion
        "nesbot/carbon": "^2.72.2|^3.0.0-beta.3@beta",
```

File: src/Illuminate/Support/composer.json [hunk 2]
```diff
@@ -23,7 +23,7 @@
         "illuminate/conditionable": "^11.0",
         "illuminate/contracts": "^11.0",
         "illuminate/macroable": "^11.0",
-        "nesbot/carbon": "^2.67",
+        "nesbot/carbon": "^2.72.2 || ^3.0.0-beta.3@beta",
```
Comment (driesvints) [inline src/Illuminate/Support/composer.json:26]:
```suggestion
        "nesbot/carbon": "^2.72.2|^3.0.0-beta.3@beta",
```

**Review by driesvints — APPROVED** [review]

**Comment by driesvints** [comment]
Thanks @kylekatarnls. Let me know as soon as you have a stable v3 out. We'll update our packages then as well.

**Comment by kylekatarnls** [comment]
Done #49940

**Pull request merged**